src/condor_utils/submit_utils.cpp

```diff
@@ -7312,16 +7312,17 @@ int SubmitHash::FixupTransferInputFiles()
 // if return value is true, but *ads_error is not empty() then the request ads have missing fields
 // that are required by configuration.
 bool SubmitHash::NeedsOAuthServices(
+	bool add_local,	// in: Add local issuer/client services mentioned in configuration
 	std::string & services,   // out: comma separated list of services names for OAuthServicesNeeded job attribute
-	ClassAdList * request_ads /*=NULL*/, // out: optional list of request classads for the services
+	std::vector<ClassAd> * request_ads /*=NULL*/, // out: optional list of request classads for the services
 	std::string * ads_error /*=NULL*/) const // out: error message from building request_ads
 {
-	if (request_ads) { request_ads->Clear(); }
+	if (request_ads) { request_ads->clear(); }
```

Comment:
As a least-astonishment thing, should this be cleared before we actually do anything with it?

Reply:
Are you suggesting we shouldn't clear the vector before storing into it?

Reply:
My observation is that the function can exit after clearing the vector but before using it for anything, and that it seems good general practice not to change things unnecessarily.

Reply:
Got it. The callers pass in an empty vector expecting it to be filled, so it's not an issue in practice. I plan to leave it be.